### PR TITLE
Add dolt-specific tests for a GMS change

### PR DESCRIPTION
**Pull request by PavelSafronov**
Adding BATS tests that will exercise the changes introduced by https://github.com/dolthub/go-mysql-server/pull/1389
Tests will fail until PR 1389 is merged and deployed.

### CODE REVIEW

**Review by fulghum — COMMENTED** [review]
Looks good! Just a couple comments, mostly brainstorming the best locations for these tests.

File: integration-tests/bats/dolt_clone.bats [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    cd $repo2
+
+    # Clone repo 1 into repo 2
+    run dolt sql -q "call dolt_clone('file://$repo1/.dolt/noms', 'repo1');"
+    echo "call dolt_clone >>> $output"
+    [ "$status" -eq 0 ]
+
+    # verify databases
+    run dolt sql -q "show databases;"
+    echo "show databases>>> $output"
```
[the remark this answers is not in the document]
Comment (PavelSafronov) [reply integration-tests/bats/dolt_clone.bats:54]: Yup, left-over debug statements, removing.

But while we're here, my approach to getting logs from BATS has been to add these echo calls and a `false` statement at the end of a test. Is there a better approach for "debugging" these tests?

File: integration-tests/bats/dolt_clone.bats [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    cd $BATS_TMPDIR
+    cd dolt-repo-$$
+    mkdir "dolt-repo-clones"
+}
+
+teardown() {
+    teardown_common
+}
+
+@test "clone: dolt_clone procedure in empty dir" {
```
[the remark this answers is not in the document]
Comment (PavelSafronov) [reply integration-tests/bats/dolt_clone.bats:16]: Initially, putting this test in its own file just allowed me to quickly re-run it. So this is another "debugging" artifact. Removing the new file and adding this test to the tail of sql-server.bats.

File: integration-tests/bats/tableplus.bats [hunk 3]
```diff
@@ -243,3 +243,50 @@ SQL
   [ "$status" -eq 0 ]
   [[ "$output" =~ "status" ]] || false
 }
+
+@test "tableplus: server with no dbs yet should be able to describe dolt stored procedures" {
```
[the remark this answers is not in the document]
Comment (PavelSafronov) [reply integration-tests/bats/tableplus.bats:247]: Yup, definitely the wrong place for these tests. Moving them to sql.bats, that seems like a reasonable spot.

**Pull request merged**